src/coreclr/vm/riscv64/stubs.cpp

```diff
@@ -1307,151 +1361,370 @@ void StubLinkerCPU::EmitMovReg(IntReg Xd, IntReg Xm)
 void StubLinkerCPU::EmitMovReg(FloatReg dest, FloatReg source)
 {
     Emit32(RTypeInstr(0x53, 0, 0x11, dest, source, source));  // fsgnj.d
+    LOG((LF_STUBS, LL_EVERYTHING, "fmv %s, %s\n", fpRegAbiNames[dest], fpRegAbiNames[source]));
 }
 
-void StubLinkerCPU::EmitSubImm(IntReg Xd, IntReg Xn, unsigned int value)
+void StubLinkerCPU::EmitSubImm(IntReg Xd, IntReg Xn, int value)
 {
-    _ASSERTE(value <= 0x800);
-    EmitAddImm(Xd, Xn, ~value + 0x1);
+    EmitAddImm(Xd, Xn, -value);
```

Comment:
Why this change? Assuming it's correct would be good to adjust comment: https://github.com/dotnet/runtime/blob/268a28b80bc8511887aeceb36e20977907e87c07/src/coreclr/vm/riscv64/stubs.cpp#L1075C12-L1075C23

Reply:
I changed the signature for consistency with other Emit* methods, all 12-bit immediates on RISC-V are signed.

> would be good to adjust comment

I folded the whole if statement.